### ISSUE TITLE
Keyboard Remap

### ISSUE THREAD

**Issue by death7654** [death7654/Chrultrabook-Tools#24]
![image](https://github.com/death7654/Chrultrabook-Controller/assets/72635727/7b12c80c-4da5-42d0-ab49-551e9f712cc4)

**Comment by death7654** [comment]
https://github.com/coolstar/VivaldiKeyboardTester

**Comment by death7654** [comment]
![image](https://github.com/death7654/Chrultrabook-Controller/assets/72635727/f53388cb-3c43-46f0-a388-b4c02588aa98)

**Comment by death7654** [comment]
![image](https://github.com/death7654/Chrultrabook-Controller/assets/72635727/f6382b83-9a68-4ea7-83e8-4f76ad5b22f4)

**Comment by death7654** [comment]
Fix vertical spacing

**Comment by death7654** [comment]
> Fix vertical spacing 
Fixed and added vertical keys

**Comment by ChocolateLoverRaj** [comment]
How would this app interact with the Windows keyboard driver settings?

**Comment by WeirdTreeThing** [comment]
Generate a new config bin.

**Comment by death7654** [comment]
<https://github.com/FyraLabs/crosec-rs>

**Comment by ChocolateLoverRaj** [comment]
> https://github.com/FyraLabs/crosec-rs

@death7654 do u want Windows support added to it?

**Comment by death7654** [comment]
I'll be doing that soon lol

**Comment by WeirdTreeThing** [comment]
> > https://github.com/FyraLabs/crosec-rs
> 
> @death7654 do u want Windows support added to it?

I had done some initial work for windows support in https://github.com/FyraLabs/crosec-rs/commit/02976a0313d51f0b9d756662085a031ef0daf5d1 but never got around to implementing it. And then you removed it in https://github.com/FyraLabs/crosec-rs/commit/614d48da0ba7dcfac1d6e8e9d9bfcbdb30f3d2f1 (fair enough, though).

**Comment by ChocolateLoverRaj** [comment]
@WeirdTreeThing @death7654 we should discuss how to make crosec more flexible and allow for adding support for non-Linux interfaces to the EC. Right now the commands depend on having a Linux fd. We could change that and make it a struct which represents an "opened" `cros_ec` or `cros_fp` and have functions like `init` or `open` for it that are different based on the platform.

**Comment by death7654** [comment]
That would be redundant as it would waste memory or space for binaries that belong to other operating systems. Not to mention the added complexity

**Comment by ChocolateLoverRaj** [comment]
> That would be redundant as it would waste memory or space for binaries that belong to other operating systems

No, because it would use `#[cfg]` and different structs for different OSs so the output binary would not have code or memory that is used in a different OS. I might not be explaining it well. I could show some example code of what I mean.

Do you have a better idea?

**Comment by death7654** [comment]
It is best to keep it seperate as what works on linux may not work on windows. For optimal quality seperation is needed

**Comment by WeirdTreeThing** [comment]
???? What rajas said is fine. The ec command structs are all the same, just the interface for them is different. On windows there is no character device to write to, instead you write to an io port. On linux we do have the character device which itself writes to that io port, but you can also directly write to that port. We can also implement spi for direct cros_fp.

**Comment by death7654** [comment]
My apologies, I did not fully read the comment as I was in class.

**Comment by death7654** [comment]
The windows version has been implemented in the applicaion, linux support will come at a later time

**Comment by death7654** [comment]
A new tracker issue for linux will be opened, windows remapping is complete
